node/config/load.go

```diff
@@ -46,7 +72,51 @@ func FromReader(reader io.Reader, def interface{}) (interface{}, error) {
 	return cfg, nil
 }
 
-func ConfigUpdate(cfgCur, cfgDef interface{}, comment bool) ([]byte, error) {
+// function from raw config string to loadability
+type loadabilityCheck struct {
+	configOk       func(string) error
+	configNotFound func() error
+}
+
+func DefaultFullNodeLoadabilityCheck() loadabilityCheck {
+	configOk := func(cfgRaw string) error {
+		poisonStr := "#EnableSplitstore =" // splitstore must be set explicitly
+		if strings.Contains(cfgRaw, poisonStr) {
+			return xerrors.Errorf("Config contains disallowed string %s, refusing to load. Please explicitly set EnableSplitstore. Set it to false to keep chain management unchanged from March 2023 set it to true if you want to discard old state by default", poisonStr)
+		}
+		return nil
+	}
+	configNotFound := func() error {
+		return xerrors.Errorf("FullNode config not found and fallback to default disallowed.  Set up config and explicitly set EnableSplitstore.  Set it to false to keep chain management unchanged from March 2023 set it to true if you want to discard old state by default ")
+	}
+	return loadabilityCheck{
+		configOk:       configOk,
+		configNotFound: configNotFound,
+	}
+}
+
+func LoadabilityCheckNone() loadabilityCheck {
+	return loadabilityCheck{
+		configOk:       func(string) error { return nil },
+		configNotFound: func() error { return nil },
+	}
+}
+
+type keepUncommented func(string) bool
+
+func DefaultFullNodeCommentFilter() keepUncommented {
+	return func(s string) bool {
+		return strings.Contains(s, "EnableSplitstore")
```

Comment:
This is really sloppy. I assume you're saying "keep the enable splitstore line uncommented"?

Ideally, we'd actually follow the structure of the config. At a _minimum_, we should be checking that this string appears at the beginning of the line (i.e., `^EnableSplitstore =`).

But there also needs to be a lot of comments/documentation explaining why we're doing something like this. It's really non-obvious.

Reply:
Its a little sloppy and I agree `^EnableSplitstore =` is better but this *does* follow the structure of the config.  When we traverse ConfigUpdate this gets applied to config field lines, not sections, not whitespace, not comments.

Reply:
ok I've added beginning of line and whitespace aware regexp imo this is good enough